blog/_posts/2018-04-04-fixing-subpath-volume-vulnerability.md

```diff
@@ -0,0 +1,193 @@
+---
+layout: blog
+title: Fixing the Subpath Volume Vulnerability in Kubernetes
+date: Wednesday, April 4, 2018
+---
+
+On March 12, 2018, the Kubernetes Product Security team disclosed [CVE-2017-1002101](https://issue.k8s.io/60813), which allowed containers using [subpath](https://kubernetes.io/docs/concepts/storage/volumes/#using-subpath) volume mounts to access files outside of the volume. This means that a container could access any file available on the host, including volumes for other containers that it should not have access to.
+
+The vulnerability has been fixed and released in the latest Kubernetes patch releases. We recommend that all users upgrade to get the fix. For more details on the impact and how to get the fix, please see the [announcement](https://groups.google.com/forum/#!topic/kubernetes-announce/6sNHO_jyBzE). Unfortunately, some functional regressions were found after the initial fix and are being tracked in [issue #61563](https://github.com/kubernetes/kubernetes/issues/61563).
+
+This post presents a technical deep dive on the vulnerability and the solution.
+
+## Kubernetes Background
+
+To understand the vulnerability, one must first understand how volume and subpath mounting works in Kubernetes.
+
+Before a container is started on a node, the kubelet volume manager locally mounts all the volumes specified in the PodSpec under a directory for that Pod on the host system. Once all the volumes are successfully mounted, it constructs the list of volume mounts to pass to the container runtime. Each volume mount contains information that the container runtime needs, the most relevant being:
+
+* Path of the volume in the container
+* Path of the volume on the host (/var/lib/kubelet/pods/<pod uid>/volumes/<volume type>/<volume name>)
```

Comment:
The directory is not getting formatted correctly because it has the "<" characters.  Can you enclosure all the directories with "`"?